### PR TITLE
create-a-container: MVC manifesto + phases 0-1 (test scaffolding, apikeys pilot migration)

### PR DESCRIPTION
**Pull request by runleveldev**
## Summary

First installment of the create-a-container MVC refactor: the manager's route files have grown into "god routes" (`routers/api/v1/containers.js` is 860 lines; its POST handler alone is ~175 lines mixing validation, business logic, eight model writes, manual transactions, and error mapping). This PR establishes the target architecture, the test safety net, and proves the pattern end-to-end on the smallest resource.

## Before you balk at the diff size

The raw diff is ~6,400 insertions. **94% of that is not application code:**

| Cut | Lines |
|---|---|
| Raw diff | +6,363 / −1,425 (25 files) |
| ...minus `package-lock.json` (~5,100), tests + helpers (~570), manifesto doc (~430), jest/Make config | **+374 / −187** (12 files) |

And within that +374 of application code:

| Category | Lines | What |
|---|---|---|
| Moved, not written | ~130 | `server.js` `main()` relocated verbatim into `app.js` (see the matching −95 in server.js) |
| New shared infrastructure | 53 | `middlewares/validate.js` (zod validation → standard error envelope) |
| apikeys resource slice | +154 / −89 | six files in `resources/apikeys/` replacing the 89-line route file |
| Genuinely new behavior | ~20 | session-sweeper `unref()`, `secure: 'auto'` cookie, app-level `csrfGuard` + `jsonErrorHandler` |
| Wiring/config | ~10 | router mount, package.json deps + test script |

Net: roughly **+65 lines of new production behavior**, plus +65 net for the apikeys migration — the growth there is the cost of layer boundaries (imports/exports/docblocks across six files instead of one) and is front-loaded: the shared scaffolding amortizes across every subsequent resource, and the big files (`containers.js`) already contain duplicated serializers and inline query builders that these layers absorb.

Suggested review order: the manifesto, then `resources/apikeys/` against the deleted `routers/api/v1/apikeys.js`, then `app.js` side-by-side with the old `server.js`, then the tests.

## The architecture

**Read first: [`create-a-container/docs/mvc-manifesto.md`](https://github.com/runleveldev/opensource-server/blob/mvc/create-a-container/docs/mvc-manifesto.md)** — normative for humans and agents. It defines:

- **Resource-first layout**: `resources/<name>/` containing `router.js`, `controller.js`, `service.js`, `repository.js`, `serializer.js`, `validator.js`, with strict downward-only dependencies. Layers that outgrow one file become folders (`service/index.js` + focused modules).
- **A frozen wire contract**: same envelope (`{ data }` / `{ error: { code, message, fields? } }`), status codes, and paths — the React client and `openapi.v1.yaml` must not notice the refactor. Built on the existing `middlewares/api.js` contract (`ApiError`, `asyncHandler`, `jsonErrorHandler`), not a replacement for it.
- **A strangler-fig migration playbook**: pin behavior with integration tests first, then extract serializer → repository → service → controller/router, one resource per PR, ordered smallest (`apikeys`) to largest (`containers`).

## What's in this PR

| Commit | What |
|---|---|
| `f7848d2f` | The manifesto |
| `4d4f7962` | Phase 0: zod + jest + supertest; shared `middlewares/validate.js`; sqlite test DB; user/API-key factories |
| `0078dd19` | 10 integration tests pinning the `/api/v1/apikeys` wire contract **before** touching it |
| `42e1b6a5` | Phase 1: `routers/api/v1/apikeys.js` (89 lines) → six-file slice in `resources/apikeys/`; router is now pure wiring; +5 service unit tests |
| `1068d51f` | Split `app.js` (buildApp) from `server.js` (migrations + secrets + listen) so tests exercise the **real** production middleware stack, not a hand-rolled copy |
| `893dbfd4` | `make test` added to the repo Makefile standard (root loop + all three components) |
| `168da729` | Verbose jest output; fix an intermittent session-store/schema race in test setup |
| `7ad1d19d` | CodeQL: `secure: 'auto'` session cookie (object form, statically analyzable); app-wide `csrfGuard` + app-level `jsonErrorHandler` |

Rebased onto current `main` (#373 package enhancements): the merged `server.js` keeps the new umzug `runMigrations` startup step and the fatal-error exit handler on the server.js side of the app/server split, and `package.json` folds `umzug` + the `sequelize-cli` devDependencies move together with the new test dependencies.

## Behavior notes

The pinned integration tests pass unchanged across the migration. Deliberate error-path changes:
- apikeys: malformed non-UUID `:id` → `400 invalid_request` (previously 404 on sqlite / would be 500 on postgres); `description` > 255 chars → `400` (previously dialect-dependent DB error)
- CSRF failures and errors on non-API surfaces now render the JSON error envelope instead of Express's default HTML page with a stack trace (app-level `jsonErrorHandler`)
- mutations to unknown paths → 403 (CSRF) instead of 404 (app-wide `csrfGuard`; all real state-changing routes live under `/api/v1` where the guard already applied)

## CodeQL status

`js/clear-text-cookie` is fixed (`secure: 'auto'`). `js/missing-token-validation` remains open as a **false positive**: CodeQL does not model `csrf-sync` (only csurf/tiny-csrf/lusca/fastify), and the synchronizer-token check runs inside the library where the analyzer doesn't look. The protection is real — `csrfGuard` wraps `csrf-sync`'s `csrfSynchronisedProtection` and now covers every handler app-wide. Will be dismissed manually.

## Conflict review

Checked all open PRs before starting: #394, #365, #318, #296 touch `containers.js`/`nodes.js`/models — none touch `apikeys`, `middlewares/`, `server.js`, or the Makefiles. The playbook sequences `nodes` after #394 and `containers` last partly for this reason.

## Testing

- `make test` from the repo root: 23 tests (10 pinned integration via supertest against the real app + sqlite, 5 service unit, 8 validate middleware), verified over 5+ consecutive runs and re-run after the rebase
- Boot smoke tests: startup migrations run, `/api/v1/health` 200, unauthenticated `/api/v1/apikeys` 401, CSRF failure returns the exact `{"error":{"code":"csrf_invalid"}}` envelope

## Next (separate PRs)

Phase 2: `settings`, `groups`, `external-domains` per the playbook. A CI workflow invoking `make test` would make the pinning tests guard those migrations — happy to add it here or separately.

### CODE REVIEW

File: create-a-container/Makefile [hunk 1]
```diff
@@ -17,7 +17,7 @@ APP_FILES := server.js job-runner.js package.json package-lock.json openapi.v1.y
 APP_DIRS  := bin config data middlewares migrations models node_modules \
              public routers seeders utils views
 
-.PHONY: help deps build dev install deb rpm apk package clean
+.PHONY: help deps build dev test install deb rpm apk package clean
```
[the remark this answers is not in the document]
Comment (runleveldev) [reply create-a-container/Makefile]: Good catch — fixed in 51ba7528. `app.js` added to `APP_FILES` and `resources` to `APP_DIRS`, and the install target now prunes `__tests__/` directories from the staged tree (middlewares/ and resources/ carry colocated tests now). Verified with `make install DESTDIR=<stage>`: both new entrypoints ship, zero `__tests__` dirs staged.

File: create-a-container/resources/apikeys/validator.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const { z } = require('zod');
+
+// Matches the ApiKeys.description column (STRING(255), nullable). Length is
+// enforced here so oversized input fails as a 400 instead of a
+// dialect-dependent DB error.
+const createApiKey = z.object({
+  description: z.string().max(255).optional(),
+});
```
[the remark this answers is not in the document]
Comment (runleveldev) [reply create-a-container/resources/apikeys/validator.js:10]: Correct — that was a wire-contract regression. Fixed in 51ba7528: the schema is now `z.string().max(255).nullish()` so an explicit JSON `null` passes through and is stored as NULL, matching the legacy router. Also added an integration test pinning `{ "description": null }` → 201 with `description: null` so it can't regress again.

**Review by cmyers-mieweb — DISMISSED** [review]
